src/apps/lwaftr/lwaftr.lua

```diff
@@ -43,6 +44,43 @@ local function compute_binding_table_by_ipv4(binding_table)
    return ret
 end
 
+local function port_to_psid(port, reserved_ports_bit_count, psid_len)
+   local shift = 16 - (reserved_ports_bit_count + psid_len)
+   return band(rshift(port, shift), (2^psid_len)-1)
+end
+
+local function compute_psid_info_map(binding_table)
+   local value_type = ffi.typeof([[
+      struct { uint16_t reserved_ports_bit_count; uint16_t psid_len; }
```

Comment:
here we should specify psid_len and shift, IMO.  i agree with widening these to 16 bits each to make the total entry size 8 bytes (with the 4 byte ipv4), excellent.